packages/kit/src/runtime/client/client.js

```diff
@@ -790,17 +785,29 @@ export function create_client({ target, base, trailing_slash }) {
 			if (loaders[i]) {
 				try {
 					branch.push(await branch_promises[i]);
-				} catch (e) {
-					const error = normalize_error(e);
-
-					if (error instanceof Redirect) {
+				} catch (err) {
+					if (err instanceof Redirect) {
 						return {
 							type: 'redirect',
-							location: error.location
+							location: err.location
 						};
 					}
 
-					const status = e instanceof HttpError ? e.status : 500;
+					let status = 500;
+					/** @type {App.PageError} */
+					let error;
+
+					if (/** @type {any} */ (err)?.type === 'error') {
+						// this is the server error rethrown above, reconstruct but don't invoke
```

Comment:
do we know this for certain? what if someone does `throw { type: 'error' }`? edge case, i know

Reply:
We don't, no. It could definetly happen that way. To be completely safe we should append something that is super unlikely to be present somewhere else, like `__kit_serverload_error`